File: mini-max-sum/mini-max-sum.clj
(defn miniMaxSum [pos-ints]
  (let
   [sums (map (fn [p-int] (- (reduce + pos-ints) p-int)) pos-ints)
    {mini :mini max :max}
    (reduce
     (fn [{mini :mini max :max} sum]
       (cond
         (< sum mini) {:mini sum :max max}
         (> sum max) {:mini mini :max sum}
         :else {:mini mini :max max}))
     {:mini (first sums) :max (first sums)}
     sums)]
    (prn mini max)))

File: mini-max-sum/mini-max-sum.js
// https://www.hackerrank.com/challenges/mini-max-sum/problem

const miniMaxSum = posIntegers => {
  const sums = posIntegers.map((posInteger, i) => {
    return posIntegers.reduce((sum, otherPosInteger, j) => {
      const isSameIndex = i === j;
      return isSameIndex ? sum : sum + otherPosInteger;
    }, 0);
  });

  let miniSum;
  let maxSum;

  sums.forEach(sum => {
    if (!miniSum) miniSum = sum;
    if (!maxSum) maxSum = sum;

    const isMiniSum = sum < miniSum;
    if (isMiniSum) miniSum = sum;

    const isMaxSum = sum > maxSum;
    if (isMaxSum) maxSum = sum;
  });

  console.log(miniSum, maxSum);
};
